RF: autoreplace assert_true(...==...) with plain assert

yarikoptic opened the pull request:
It is desired since `assert_true` (not even assert_equal) provides no information on actual value which fails the test

- [x] verify/fix any possible incorrect substitution

Note: Command (`git-sedi`) recorded in the [commit msg](https://github.com/scikit-learn/scikit-learn/commit/ef7a8dc5982d6006bcfffef93e736b59ba5b0f39) comes from my [zshrc](http://git.onerussian.com/?p=etc/zsh.git;a=blob;f=.zsh/zshrc/30_aliases;h=6497fed798440760540a4296df6d4f4ee89be9d2;hb=HEAD#l110):
```shell
# Useful for quick refactoring
git-sedi() {
  git grep -l $1 | xargs sed -i -e "s/$1/$2/g"
}
```

yarikoptic committed:
[DATALAD RUNCMD] RF: autoreplace assert_true(...==...) with plain assert

It is desired since assert_true (not even assert_equal) provides no information
on actual value which fails the test

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "git-sedi 'assert_true(\\( *.* == .*\\) *)$' 'assert \\1'",
 "exit": 0,
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

amueller commented:
invaid syntax ;)

amueller commented on sklearn/model_selection/tests/test_search.py:
```diff
@@ -752,14 +752,14 @@ def check_cv_results_array_types(search, param_keys, score_keys):
     assert_array_equal(sorted(cv_results.keys()),
                        sorted(param_keys + score_keys + ('params',)))
-    assert_true(all(cv_results[key].shape == (n_cand,)
+    assert all(cv_results[key].shape == (n_cand,
```
invalid syntax here I think?

eamanu replied:
also I see that the assert change from the original, is not?

yarikoptic replied:
yeah -- the 'cons' of auto sed. will need to fix them up - didn't spot on a brief eye balling

yarikoptic committed:
BF: fixed syntax error introduced by automatic substitution

yarikoptic committed:
[DATALAD RUNCMD] git-sedi 'assert_true( *\(.*\) *) *$' 'a...

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "git-sedi 'assert_true( *\\(.*\\) *) *$' 'assert \\1'",
 "exit": 0,
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
BF: fixed up some incorrect replacements (due to multiline) for assert_true

yarikoptic committed:
FLAKE8: removed now present over indentation (no _true any longer)

qinhanmin2014 reviewed (approved):
We also need to remove relevant imports

rth reviewed (commented):
+1 for removing unused imports, we could also do the same thing for `assert_equal` in a follow-up PR. 

I would even suggest including this in 0.20.1 as it shouldn't affect users, but make understanding test failure logs easier.

There was an earlier discussion about the possibility of creating merge conflicts (can't find it anymore), but in this case, I think it's worth it.

qinhanmin2014 commented:
Fine with 0.20.1, as long as we can finish it.

@rth Maybe we don't need to do the same thing for assert_equal? We can still get things like ``AssertionError: a != b`` if we use assert_equal.

amueller commented:
I don't think there's a very strong incentive for assert_equal, so I wouldn't prioritize it?

amueller reviewed (approved):
would be nice to remove unused imports but that's manual work so I'd be fine with a separate PR.

stsouko commented:
```assert_true(all([not isinstance(est, RandomForestClassifier) for est in
                     eclf2.estimators_]))
```
checks like this not fixed.

jnothman commented:
What's hard about handling the message? The assert statement allows for a
message

qinhanmin2014 commented:
> What's hard about handling the message? The assert statement allows for a
message

You're right, apologies for the stupid comment.

@stsouko Would you like to submit a PR? Otherwise I'll open an issue.

stsouko commented:
At this moment I did a #12513. This assertion refers to the tests of voting classifier module. I fixed this test in PR.

qinhanmin2014 commented:
@stsouko please do it in a separate PR, thanks

stsouko commented:
I can do full fix of voting tests. according to [this](https://github.com/scikit-learn/scikit-learn/pull/12513#discussion_r232090187) conclusion. It is OK?

qinhanmin2014 commented:
It's better to fix a single issue in each PR.

stsouko commented:
done.
